src/VisualStudio/Core/Def/Implementation/DesignerAttribute/DesignerAttributeIncrementalAnalyzer.cs

```diff
@@ -200,49 +210,59 @@ private void RegisterDesignerAttribute(Document document, string designerAttribu
                 return;
             }
 
-            var documentId = document.Id;
-            _notificationService.RegisterNotification(() =>
+            var updateService = await GetUpdateServiceIfCpsProjectAsync(document.Project, cancellationToken).ConfigureAwait(false);
+            if (updateService != null)
+            {
+                // we track work by async token but doesn't explicitly wait for it
+                var asyncToken = _listener.BeginAsyncOperation("RegisterDesignerAttribute");
+                _ = updateService.SetProjectItemDesignerTypeAsync(document.FilePath, designerAttributeArgument).CompletesAsyncOperation(asyncToken);
```

Comment:
Doesn't this drop exceptions in the task?

Reply:
@Pilchie want me to fire NFW? I can put continue with that can save exception if there is one.

Reply:
That seems reasonable to me.